pkg/package-server/provider/registry.go

```diff
@@ -189,7 +189,18 @@ func (p *RegistryProvider) syncCatalogSource(obj interface{}) (syncError error)
 		Namespace: source.GetNamespace(),
 		Name:      source.GetName(),
 	}
+
 	if sourceMeta := p.sources.GetMeta(key); sourceMeta != nil && sourceMeta.Address == address {
+		logger.Infof("updating PackageManifest based on CatalogSource changes: %v", key)
+		timeout, cancel := context.WithTimeout(context.Background(), cacheTimeout)
+		defer cancel()
+		client, err := p.registryClient(key)
+		if err != nil {
+			syncError = err
+			return
```

Comment:
```suggestion
		client, syncError := p.registryClient(key)
		if syncErr != nil {
			return
		}
```

Reply:
Thanks! I will update once the dependent PR is merged!

Reply:
Just realized, i can't do this, and was delieberatly avoiding it. This reinitialized `syncError`.

Reply:
Can we just `return err` on line 199?

Reply:
@Bowenislandsong just change `:=` to `=` and decalre the client variable before the function call